tests/phpunit/tests/rest-api.php

```diff
@@ -2558,4 +2558,29 @@ public function test_route_args_is_array_of_arrays() {
 
 		$this->assertTrue( $registered );
 	}
+
+	/**
+	 * @ticket 62932
+	 */
+	public function test_should_return_error_if_rest_route_not_string() {
+		global $wp;
+
+		$wp = new stdClass();
+
+		$wp->query_vars = array(
+			'rest_route' => array( 'invalid' ),
+		);
+
+		$this->expectException( WPDieException::class );
+
+		try {
+			rest_api_loaded();
+		} catch ( WPDieException $e ) {
+			$this->assertStringContainsString(
+				'The rest route parameter must be a string.',
+				$e->getMessage()
+			);
+			throw $e; // Re-throw to satisfy expectException
+		}
```

Comment:
You can simplify the test by replacing it with the following:

* `go_to()` is a helper used in a lot of locations to set up the `$wp` global, using it here will assist future contributors reading the code.
* TIL that `expectExceptionMessageMatches` exists (for older versions of PHPUnit WP has it polyfilled via the Yoast\PHPUnitPolyfills package).

```suggestion
		$this->expectException( WPDieException::class );
		$this->expectExceptionMessageMatches( '/^The rest route parameter must be a string.$/' );
		$this->go_to( '/?rest_route[]=invalid' );
		rest_api_loaded();
```

Reply:
I don't have much experience with unit tests but this change looks much cleaner and better, you can commit this at your convenience  🚀